master/buildbot/test/fake/fakedb.py

```diff
@@ -1655,7 +1656,7 @@ def setState(self, objectid, name, value):
 
     def atomicCreateState(self, objectid, name, thd_create_callback):
         value = thd_create_callback()
-        self.states[objectid][name] = json.dumps(value)
+        self.states[objectid][name] = json.dumps(bytes2NativeString(value))
```

Comment:
that is also the case for  

    json.dumps({'a': b'b'})

but bytes2NativeString will not handle that case.

How many callers have that issue?

Normally, the contract for using db.state is that the input is json serializable.

So I would say the bytes2NativeString should be done by the caller.
And we should make sure that the caller is okay to receive native string when the data is deserialized.